search/search_repository.go

```diff
@@ -382,11 +371,66 @@ func (q Query) determineLiteralType(key string, val string) criteria.Expression
 func (q Query) generateExpression() (criteria.Expression, error) {
 	var myexpr []criteria.Expression
 	currentOperator := q.Name
-	if !isOperator(currentOperator) {
-		var key string
-		if val, ok := searchKeyMap[q.Name]; ok {
-			key = val
+
+	// Here we handle the "$WITGROUP" query parameter which we translate from a
+	// simple
+	//
+	// "$WITGROUP = y"
+	//
+	// expression into an
+	//
+	// "Type in (y1, y2, y3, ... ,yn)"
+	//
+	// expression where yi represents the i-th work item type associated with
+	// the work item type group y.
+	handleWitGroup := func(q Query) error {
+		typeGroupName := q.Value
+
+		if typeGroupName == nil {
+			return errors.NewBadParameterError(WITGROUP, typeGroupName).Expected("not nil")
+		}
+		typeGroup := workitem.TypeGroupByName(*typeGroupName)
+		if typeGroup == nil {
+			return errors.NewBadParameterError(WITGROUP, *typeGroupName).Expected("existing " + WITGROUP)
+		}
+		var e criteria.Expression
+		if !q.Negate {
+			for _, witID := range typeGroup.TypeList {
+				eq := criteria.Equals(
+					criteria.Field("Type"),
+					criteria.Literal(witID.String()),
+				)
+				if e != nil {
+					e = criteria.Or(e, eq)
+				} else {
+					e = eq
+				}
+			}
 		} else {
+			for _, witID := range typeGroup.TypeList {
+				eq := criteria.Not(
+					criteria.Field("Type"),
+					criteria.Literal(witID.String()),
+				)
+				if e != nil {
+					e = criteria.And(e, eq)
+				} else {
+					e = eq
+				}
+			}
+		}
+		myexpr = append(myexpr, e)
+		return nil
+	}
+
+	if q.Name == WITGROUP {
```

Comment:
I found `currentOperator == WITGROUP` more readable, that variable is there already

Reply:
To be honest, I didn't see that. Thank you. But I must say that `currentOperator` is a completely stupid word because in some cases the "operator" is just a field name. So why change the query concept here?

Reply:
And btw. it only really is an operator if `isOperator` returns `true`.